langchain/chains/combine_documents/map_reduce.py

```diff
@@ -49,14 +55,79 @@ def get_default_document_variable_name(cls, values: Dict) -> Dict:
                 )
         return values
 
-    def combine_docs(self, docs: List[Document], **kwargs: Any) -> str:
-        """Combine by mapping first chain over all, then stuffing into final chain."""
+    def compute_tokens(self, results: List, docs: List[Document], **kwargs: Any) -> int:
+        """Checks to see if, after combining a set of documents into one text, if it is too long."""
+        question_result_key = self.llm_chain.output_key
+        result_docs = [
+            Document(page_content=r[question_result_key], metadata=docs[i].metadata)
+            # This uses metadata from the docs, and the textual results from `results`
+            for i, r in enumerate(results)
+        ]
+        doc_dicts = []
+        for doc in result_docs:
```

Comment:
seems like it would be generically useful to expose a method on all combine document chains that gets the prompt for the passed in documents

Reply:
Agreed.